### Data Set: Tokyo
Visualización de datos del data set Tokyo.</br>
Exportación de librerias de Visualización del data set:

In [1]:
from pandasgui import show
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.express as px


#plt.style.use("seaborn")

### Tipo de datos dentro del dataset

#### Tenemos las siguientes columnas:

- **id:** Un identificador único para cada registro en tu conjunto de datos. Este campo generalmente se utiliza para identificar de manera única cada entrada en el conjunto de datos.

- **name:** El nombre o título de la propiedad o alojamiento que se está listando en una plataforma como Airbnb. Este campo proporciona información sobre cómo se describe la propiedad.

- **host_id:** El identificador único del anfitrión que está listando la propiedad. Cada anfitrión tiene un número de identificación único.

- **host_name:** El nombre del anfitrión que está listando la propiedad. Este campo proporciona información sobre quién es el anfitrión.

- **neighbourhood_group:** La agrupación o categorización de vecindarios más grande o de nivel superior en una ciudad o área geográfica. Esto puede utilizarse para organizar los vecindarios en grupos más grandes para análisis.

- **neighbourhood:** El nombre del vecindario o área geográfica específica en la que se encuentra la propiedad. Esto proporciona información sobre la ubicación de la propiedad.

- **latitude:** La latitud geográfica de la ubicación de la propiedad. La latitud se utiliza para determinar la posición norte-sur en la Tierra.

- **longitude:** La longitud geográfica de la ubicación de la propiedad. La longitud se utiliza para determinar la posición este-oeste en la Tierra.

- **room_type:** El tipo de habitación o alojamiento que se está listando, como "habitación privada", "casa/apartamento completo" o "habitación compartida".

- **price:** El precio de alquiler por noche o por estancia en la propiedad. Este campo indica cuánto cuesta alojarse en la propiedad.

- **minimum_nights:** El número mínimo de noches que un huésped debe reservar para quedarse en la propiedad. Esto puede variar según la política del anfitrión.

- **number_of_reviews:** El número total de revisiones o comentarios que ha recibido la propiedad hasta la fecha. Esto puede proporcionar información sobre la popularidad y la calidad de la propiedad.

- **last_review:** La fecha de la última revisión o comentario dejado por un huésped en la propiedad. Esto puede ayudar a determinar cuándo fue la última vez que alguien se quedó en la propiedad.

- **reviews_per_month:** El promedio de revisiones o comentarios dejados por los huéspedes por mes. Esto puede indicar la tasa de actividad y el nivel de satisfacción de los huéspedes a lo largo del tiempo.


In [2]:
df = pd.read_csv("../data/tokyo_airbnb.csv")
df.head(2)

id                                               name  host_id  \
0   35303  La Casa Gaienmae C Harajuku, Omotesando is nearby   151977   
1  197677                          Oshiage Holiday Apartment   964081   

         host_name  neighbourhood_group neighbourhood  latitude  longitude  \
0           Miyuki                  NaN    Shibuya Ku  35.67152  139.71203   
1  Yoshimi & Marek                  NaN     Sumida Ku  35.71721  139.82596   

         room_type  price  minimum_nights  number_of_reviews last_review  \
0     Private room   4196              28                 18  2018-07-28   
1  Entire home/apt  10975               3                156  2019-05-21   

   reviews_per_month  
0               0.20  
1               1.65

In [3]:
show (df)

PandasGUI INFO — pandasgui.gui — Opening PandasGUI
C:\Users\psych\AppData\Local\Temp\ipykernel_11116\603652372.py:1: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  show (df)
C:\Users\psych\AppData\Local\Temp\ipykernel_11116\603652372.py:1: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  show (df)
C:\Users\psych\AppData\Local\Temp\ipykernel_11116\603652372.py:1: FutureWarning:

Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`

C:\Users\psych\AppData\Local\

Aplicamos una función para saber: 
Dataset tiene 14 columnas y 11466 fila.
- Valores únicos
- Ausencia de valores 
- Duplicado
- Tipo de valores

In [4]:
def summary(df):
    total_values = df.shape[0]
    missing_data = df.isnull().sum()
    missing_percentage = (missing_data / total_values) * 100
    print(f"Dataset has {df.shape[1]} features and {df.shape[0]} rows.")
    summary = pd.DataFrame(index=df.columns)
    summary["Unique"] = df.nunique().values
    summary["Missing"] = df.isnull().sum().values
    summary['Missing %'] = ((missing_data / total_values) * 100).round(2)
    summary["Duplicated"] = df.duplicated().sum()
    summary["Types"] = df.dtypes
    return summary

summary(df)

Dataset has 14 features and 11466 rows.


Unique  Missing  Missing %  Duplicated    Types
id                    11466        0       0.00           0    int64
name                  10810        0       0.00           0   object
host_id                2954        0       0.00           0    int64
host_name              2054       16       0.14           0   object
neighbourhood_group       0    11466     100.00           0  float64
neighbourhood            56        0       0.00           0   object
latitude               7295        0       0.00           0  float64
longitude              8140        0       0.00           0  float64
room_type                 3        0       0.00           0   object
price                   371        0       0.00           0    int64
minimum_nights           30        0       0.00           0    int64
number_of_reviews       251        0       0.00           0    int64
last_review             413     1677      14.63           0   object
reviews_per_month       692     1677      14.63           0  float64

Hayamos el porcentaje de peso en nulos/ausencia del dataset. 
Observamos que la columna neighbourhood_group tiene un 100% de ausencia de datos.
Las columnas: 
last_review             14.63% 
reviews_per_month       14.63%

Veremos más adelante que problemática nos pueden surgir y si son útiles

In [5]:
def summary(df):
    total_values = df.shape[0]
    missing_data = df.isnull().sum()
    missing_percentage = (missing_data / total_values) * 100
    print(f"Dataset has {df.shape[1]} features and {df.shape[0]} rows.")
    summary = pd.DataFrame(index=df.columns)
    summary["Unique"] = df.nunique().values
    summary["Missing"] = df.isnull().sum().values
    summary['Missing %'] = ((missing_data / total_values) * 100).round(2)
    summary["Duplicated"] = df.duplicated().sum()
    summary["Types"] = df.dtypes
    return summary

summary(df)

Dataset has 14 features and 11466 rows.


Unique  Missing  Missing %  Duplicated    Types
id                    11466        0       0.00           0    int64
name                  10810        0       0.00           0   object
host_id                2954        0       0.00           0    int64
host_name              2054       16       0.14           0   object
neighbourhood_group       0    11466     100.00           0  float64
neighbourhood            56        0       0.00           0   object
latitude               7295        0       0.00           0  float64
longitude              8140        0       0.00           0  float64
room_type                 3        0       0.00           0   object
price                   371        0       0.00           0    int64
minimum_nights           30        0       0.00           0    int64
number_of_reviews       251        0       0.00           0    int64
last_review             413     1677      14.63           0   object
reviews_per_month       692     1677      14.63           0  float64

Observamos outliers en Entire home/apt

In [6]:
fig = px.scatter(data_frame=df, x='room_type', y='price', color=None,
    symbol=None, size=None, trendline=None, marginal_x=None, marginal_y=
    None, facet_row=None, facet_col=None, render_mode='auto', )
fig.update()
fig.show()

In [7]:

fig = px.histogram(data_frame=df, x=['neighbourhood'],facet_row='room_type')
fig.update_layout(height=800)
fig.show()

In [8]:
df['room_type'].value_counts()

room_type
Entire home/apt    7463
Private room       3004
Shared room         999
Name: count, dtype: int64